standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/TableCacheObjects.java

```diff
@@ -3,59 +3,21 @@
 import org.apache.hadoop.hive.metastore.api.AggrStats;
 import org.apache.hadoop.hive.metastore.api.ColumnStatistics;
 import org.apache.hadoop.hive.metastore.api.Partition;
-import org.apache.hadoop.hive.metastore.api.SQLForeignKey;
-import org.apache.hadoop.hive.metastore.api.SQLNotNullConstraint;
-import org.apache.hadoop.hive.metastore.api.SQLPrimaryKey;
-import org.apache.hadoop.hive.metastore.api.SQLUniqueConstraint;
+import org.apache.hadoop.hive.metastore.api.SQLAllTableConstraints;
 
 import java.util.List;
 
 /*
  * Holder class for table objects like partitions, statistics, constraints etc.
  */
 public class TableCacheObjects {
-  private List<SQLPrimaryKey> primaryKeys;
-  private List<SQLForeignKey> foreignKeys;
-  private List<SQLNotNullConstraint> notNullConstraints;
-  private List<SQLUniqueConstraint> uniqueConstraints;
+  private SQLAllTableConstraints tableConstraints;
```

Comment:
Here tableConstraints is just a place holder to keep all constraint at one place. In cache read and write flow each constraint is treated independently. Maintained a fail safe flow by falling back to rawstore in case of dirty read, wrong values and partial info present in cached store.